### PR TITLE
Make listitem accessible

### PR DESCRIPTION
**Pull request by acrognale**

### CODE REVIEW

File: spec/root.js [hunk 1]
```diff
@@ -69,7 +69,32 @@ class Root extends Component {
           This will content filter and indexes for examples
         </NavDrawer>
 
-        <Tabs />
+        <Panel className={style.app}>
```
Comment (jodonnell) [inline spec/root.js]: whats this?

File: components/list/ListItem.js [hunk 2]
```diff
@@ -80,7 +86,14 @@ const factory = (ripple, ListItemLayout, ListItemContent) => {
       const children = this.groupChildren();
       const content = <ListItemLayout theme={theme} {...children} {...other} />;
       return (
-        <li className={`${theme.listItem} ${className}`} onClick={this.handleClick} onMouseDown={onMouseDown} onTouchStart={onTouchStart}>
+        <li
```
Comment (jodonnell) [inline components/list/ListItem.js]: this is a start but the list item is pretty broken in our app.  also if the contents has an a tag in there then you need to tab over it twice and only the inner tab will trigger the link move.

are we doing the aria stuff in a different commit?

File: components/list/ListItem.js [hunk 3]
```diff
@@ -80,7 +86,14 @@ const factory = (ripple, ListItemLayout, ListItemContent) => {
...
+          className={`${theme.listItem} ${className}`}
+          onClick={this.handleClick}
+          onMouseDown={onMouseDown}
+          onTouchStart={onTouchStart}
+          onKeyDown={this.handleEnter}
+          tabIndex={0}
+        >
           {to ? <a href={this.props.to}>{content}</a> : content}
           {children.ignored}
         </li>
```
Comment (kimberlysuazo) [inline components/list/ListItem.js]: the handleEnter should account for when there is a `to` prop added. The `<a>` it creates, receives keyboard focus, so an user would have to tab twice to get in. If we are removing the tabIndex from the `<a>`, we need to separately address how screen reader users will be aware of the link action.

Comment (acrognale) [reply components/list/ListItem.js]: This is fixed now I believe

File: components/list/ListItem.js [hunk 4]
```diff
@@ -88,8 +94,10 @@ const factory = (ripple, ListItemLayout, ListItemContent) => {
           onClick={this.handleClick}
           onMouseDown={onMouseDown}
           onTouchStart={onTouchStart}
+          onKeyDown={this.handleEnter}
+          tabIndex={to ? -1 : 0}
         >
-          {to ? <a href={this.props.to}>{content}</a> : content}
+          {to ? <a tabIndex={0} href={this.props.to}>{content}</a> : content}
```
Comment (kimberlysuazo) [inline components/list/ListItem.js]: by making the `tabIndex` of the listitem -1, we lose all of the alt text or aria attributes for that element. we want tabinex of `<a>` to be -1, and the on enter to navigate to the provided `to` link

**Pull request merged**